modules/internal/ChapelDomain.chpl

```diff
@@ -1357,9 +1357,24 @@ module ChapelDomain {
     }
 
     /*
-      This manager is returned by ``unsafeAssign()`` and can be used in
-      managed blocks to help resize arrays of non-default-initializable
-      elements.
+      An instance of this type is a context manager that can be used in
+      manage statements to resize arrays of non-default-initializable
+      element types after resizing their underlying domain.
+
+      Using an instance of this type in a manage statement will cause domain
+      assignment to occur before executing the statement body. The
+      left-hand-side of the assignment is the receiver domain that had
+      ``unsafeAssign()`` called on it, while the right-hand-side is the
+      `dom` formal of the same call.
+
+      If the assignment adds new indices to the assigned domain, then
+      corresponding elements are added to arrays declared over it.
+      If an array's element type is non-default-initializable, then any
+      newly added elements remain uninitialized.
+
+      The ``initialize()`` method can be used within the manage statement
+      body to initialize new elements of non-default-initializable arrays
+      declared over the assigned domain.
```

Comment:
Add "The new elements of the default-initializable arrays over the assigned domain will be default-initialized. They can be set to desired values as usual, for example using an assignment operator."

BTW this handling excludes `const` arrays of default-initializable elements, bummer.